core/adapters/bridge.go

```diff
@@ -31,59 +31,82 @@ func (ba *Bridge) Perform(input models.RunResult, store *store.Store) models.Run
 	} else if input.Status.PendingBridge() {
 		return resumeBridge(input)
 	}
-	ba.handleNewRun(&input, store.Config.BridgeResponseURL())
-	return input
+	return ba.handleNewRun(input, store.Config.BridgeResponseURL())
 }
 
 func resumeBridge(input models.RunResult) models.RunResult {
 	input.Status = models.RunStatusInProgress
 	return input
 }
 
-func (ba *Bridge) handleNewRun(input *models.RunResult, bridgeResponseURL *url.URL) {
+func (ba *Bridge) handleNewRun(input models.RunResult, bridgeResponseURL *url.URL) models.RunResult {
 	if ba.Params == nil {
 		ba.Params = new(models.JSON)
 	}
 	var err error
 	if input.Data, err = input.Data.Merge(*ba.Params); err != nil {
-		input.SetError(baRunResultError("handling data param", err))
-		return
+		var output models.RunResult
+		output.Data = input.Data
+		output.SetError(baRunResultError("handling data param", err))
+		return output
 	}
 
 	responseURL := bridgeResponseURL
 	if *responseURL != *zeroURL {
 		responseURL.Path += fmt.Sprintf("/v2/runs/%s", input.CachedJobRunID)
 	}
+
 	body, err := ba.postToExternalAdapter(input, responseURL)
 	if err != nil {
-		input.SetError(baRunResultError("post to external adapter", err))
-		return
+		var output models.RunResult
+		output.Data = input.Data
+		output.SetError(baRunResultError("post to external adapter", err))
+		return output
 	}
 
-	err = responseToRunResult(body, input)
-	if err != nil {
-		input.SetError(err)
-		return
-	}
+	return responseToRunResult(body, input)
 }
 
-func responseToRunResult(body []byte, input *models.RunResult) error {
+func responseToRunResult(body []byte, input models.RunResult) models.RunResult {
+	var output models.RunResult
+
 	var brr models.BridgeRunResult
 	err := json.Unmarshal(body, &brr)
 	if err != nil {
-		return baRunResultError("unmarshaling JSON", err)
+		output.Data = input.Data
+		output.SetError(baRunResultError("unmarshaling JSON", err))
+		return output
+	}
+
+	// Special case where the error comes to us via the JSON payload itself, rather than as a Go error.
```

Comment:
Maybe refer to the 'Go error' as something different?  The term makes it sound like a language specific error, as opposed to one due to local processing.